lib/guest_installation_and_configuration_base.pm

```diff
@@ -1042,23 +1227,121 @@ iptables -X
 $_iptables_default_route_devices
 iptables --append FORWARD --in-interface $_guest_network_device -j ACCEPT
 sysctl -w net.ipv4.ip_forward=1
+sysctl -w net.ipv4.conf.all.forwarding=1
 sysctl -w net.ipv6.conf.all.forwarding=1
 iptables-save > $self->{guest_log_folder}/iptables_after_modification_by_$self->{guest_name}
 EOF
 ");
     assert_script_run("chmod 777 $_network_policy_config_file");
     record_info("Network policy config file", script_output("cat $_network_policy_config_file", proceed_on_failure => 0));
     script_run("$_network_policy_config_file", timeout => 60);
-    if (script_output("cat $common_log_folder/root_cron_job | grep -i $_network_policy_config_file", proceed_on_failure => 1) eq '') {
+    $self->schedule_tasks_on_boot(_task => "$_network_policy_config_file");
+    return $self;
+}
+
+=head2 schedule_tasks_on_boot
+
+  schedule_tasks_on_boot($self, _task => $task)
+
+Schedule tasks to be executed on system boot up, please refer to these documents:
+https://docs.oracle.com/en/learn/oracle-linux-crontab/ for using crontab utility
+and https://linuxconfig.org/how-to-schedule-tasks-with-systemd-timers-in-linux for 
+for using systemd service and timer. In order to schedule a task successfully, the
+_task argument should not be empty.
+
+=cut
+
+sub schedule_tasks_on_boot {
+    my ($self, %args) = @_;
+
+    $self->reveal_myself;
+    $args{_task} //= '';
+
+    croak("Jobs to be scheduled should have _task arguments set.") if ($args{_task} eq '');
+    if (is_transactional) {
+        $self->schedule_tasks_on_boot_systemd(%args);
```

Comment:
At current stage, I still prefer to keep both and use cron on traditional system (including upcoming 15-SP5), because it had been already tested thoroughly. I think they will be kept like this for a while as long as SLES still support it. In the future, systemd might be adopted more widely over time.

Reply:
I changed to use 
```
if (script_run('systemctl is-active cron.service') != 0)
```
to differentiate systems with cron from those without.
But I will keep both systemd and cron.